primitives/src/constants.rs

```diff
@@ -28,6 +28,12 @@ pub mod currency {
 	pub fn deposit(items: u32, bytes: u32) -> Balance {
 		items as Balance * 2 * DOLLARS + (bytes as Balance) * 30 * MILLICENTS
 	}
+
+	// Value on the right side of this condition represents balance deposited to liquidity mining's pot
+	// account in native currency to prevent dusting of the pot. Pot exists for every instance of warehouse
+	// liq. mining and this value has to be deposited to all pot's instances.
+	// WARN: More tokens must be sent to pots when this value is changed.
+	static_assertions::const_assert!(NATIVE_EXISTENTIAL_DEPOSIT < 100 * UNITS);
```

Comment:
I would write an integration test for this

Reply:
Why, what would be benefits of integration test? 
This only needs to be checked when you are changing existential deposit constant and this way it will fail immediately, without `cargo test`

Reply:
1. I wouldn't expect a test in `constants.rs`
2. in an integration test, you can directly compare the existential deposit to the balance of the pot account. The existential deposit doesn't have to be less than `100*UNITS`. The balance of the pot account needs to be greater than the ED.
3. now when I'm thinking about it, it actually doesn't test anything, we are just comparing two values. It doesn't test that the amount is actually deposited to the pot account. So this test is probably not necessary.
4. I don't see a reason why compilation should fail because of this. And if it fails, we can easily fix it by increasing the number on the right side. But it actually doesn't fix anything.

Reply:
Let's ask @enthusiastmartin what is his opinion on this.

Reply:
> 1. I wouldn't expect a test in `constants.rs`
> 2. in an integration test, you can directly compare the existential deposit to the balance of the pot account. The existential deposit doesn't have to be less than `100*UNITS`. The balance of the pot account needs to be greater than the ED.
> 3. now when I'm thinking about it, it actually doesn't test anything, we are just comparing two values. It doesn't test that the amount is actually deposited to the pot account. So this test is probably not necessary.
> 4. I don't see a reason why compilation should fail because of this. And if it fails, we can easily fix it by increasing the number on the right side. But it actually doesn't fix anything.

1.- it's not "test", it's just configuration "assert".

2.- you can't directly compare it. You would have to still do transfer of 100 UNITS to `pot` account first. "default" balance(100 HDX) in the `pot` is done on chain by doing transfer to the pot from treasury when we first time create LM. ED have to be less than `100*units`, you can't use on chain balance in the `pot` because it may contain unpaid rewards e.g.: pot balance: 100(default) + 10(unpaid) and you will set ED to 101 - pot will be dusted when 10(unpaid) rewards will be paid.  

3. - yes, it doesn't test if 100 was deposited to the `pot`. We can't test it without on chain data and this test wouldn't be useful because what I wrote in 2-nd.
This test basically saying: if you set ED > 100 * UNIT, you have to change `100 * UNIT` and transfer additional amount to `pot`.

4. - compilation should fail to notify you that you have to do additional on chain action.  You always have to make sure ED < "default pot balance" and if you want to set higher ED than current limit you need to add tokens to pot.

Reply:
1. but you don't know the balance of the pot account, so there is nothing to assert. That's why I believe it's not a configuration test.
4. we have tests for this. I still don't see a reason why compilation should fail.

Reply:
Personally, i don't like this to be where it is.  It is little bit confusing. 

Wouldn't be integrity test in a corresponding pallet a better place ? 

that's usually a place where you ensure the configuration of a pallet makes sense.

Reply:
hm, the problem is that the pallet is in the warehouse so  passing `native_existential_deposit` into it is not straightforward and changing ED to > 100 would result in the necessity to upgrade integrity test in warehouse pallet + it would lock both chains to the same max ED.

FYI this is how it's done in the basilisk.

Reply:
You can get the ED from `MultiCurrency::minimum_balance(native_asset_id)`.

Reply:
> You can get the ED from `MultiCurrency::minimum_balance(native_asset_id)`.

Thank you but my bigger issue is second part